app/Http/Controllers/BlogCatController.php

```diff
@@ -34,7 +33,52 @@ public function postblogcat(Request $request)
         $cat = new CategoryPost;
         $cat -> name = $request -> name;
         $cat -> save();
-
         return redirect('addblogcat') -> with('noti', 'success');
     }
+
+    public function getEditBlogCat($id)
+    {
+        try 
+        {
+            $cat = CategoryPost::findOrFail($id);
+        } 
+        catch (ModelNotFoundException $e) 
+        {
+            echo $e->getMessage();
+        }
+        return view('backend.blogcats.editblogcat', ['cat' => $cat]);
+    }
+
+    public function postEditBlogCat(Request $request, $id)
+    {
+        try 
+        {
+            $cat = CategoryPost::findOrFail($id);
+        } 
+        catch (ModelNotFoundException $e) 
+        {
+            echo $e->getMessage();
+        }
+
+        $this->validate($request,
+        [
+            'name' => '|required|min:3|max:100'
+        ],
+        [   
+            'name.required' => trans('message.cannotblank'),
+            'name.min' => trans('message.tooshort'),
+            'name.max' => trans('message.toolong'),
+        ]);
+
+        $cat -> name = $request -> name;
```

Comment:
dấu `->` ko phải là toán tử nên không có khaorng trắng trước sau nhé
nó là phép gọi function thôi